tests/Games/NexusMods.Games.StardewValley.Tests/DependencyDiagnosticEmitterTests.cs

```diff
@@ -0,0 +1,107 @@
+using FluentAssertions;
+using Microsoft.Extensions.DependencyInjection;
+using NexusMods.Abstractions.Diagnostics;
+using NexusMods.Abstractions.Loadouts;
+using NexusMods.Abstractions.NexusWebApi.Types.V2;
+using NexusMods.Games.StardewValley.Emitters;
+using NexusMods.Games.TestFramework;
+using NexusMods.MnemonicDB.Abstractions.ElementComparers;
+using NexusMods.StandardGameLocators.TestHelpers;
+using Xunit.Abstractions;
+
+namespace NexusMods.Games.StardewValley.Tests;
+
+[Trait("RequiresNetworking", "True")]
+public class DependencyDiagnosticEmitterTests : ALoadoutDiagnosticEmitterTest<DependencyDiagnosticEmitterTests, StardewValley, DependencyDiagnosticEmitter>
+{
+    public DependencyDiagnosticEmitterTests(ITestOutputHelper testOutputHelper) : base(testOutputHelper) { }
+
+    protected override IServiceCollection AddServices(IServiceCollection services)
+    {
+        return base.AddServices(services)
+            .AddStardewValley()
+            .AddUniversalGameLocator<StardewValley>(new Version("1.6.14"));
+    }
+
+    [Fact]
+    public async Task Test_MissingRequiredDependency()
+    {
+        var loadout = await CreateLoadout();
+
+        // SMAPI 4.1.10 (https://www.nexusmods.com/stardewvalley/mods/2400?tab=files)
+        await InstallModFromNexusMods(loadout, ModId.From(2400), FileId.From(119630));
+
+        // Farm Type Manager 1.24.0 (https://www.nexusmods.com/stardewvalley/mods/3231?tab=files)
+        await InstallModFromNexusMods(loadout, ModId.From(3231), FileId.From(117244));
+
+        var diagnostic = await GetSingleDiagnostic(loadout);
+        var missingRequiredDependencyMessageData = diagnostic.Should().BeOfType<Diagnostic<Diagnostics.MissingRequiredDependencyMessageData>>(because: "Content Patcher is required for Farm Type Manager").Which.MessageData;
+        missingRequiredDependencyMessageData.MissingDependencyModId.Should().Be("Pathoschild.ContentPatcher", because: "Farm Type Manager requires Content Patcher");
+
+        // Content Patcher 2.5.3 (https://www.nexusmods.com/stardewvalley/mods/1915?tab=files)
+        await InstallModFromNexusMods(loadout, ModId.From(1915), FileId.From(124659));
+
+        await ShouldHaveNoDiagnostics(loadout, because: "The required dependency Content Patcher has been installed");
+
+        await VerifyDiagnostic(diagnostic);
+    }
+
+    [Fact]
+    public async Task Test_DisabledRequiredDependencyMessageData()
```

Comment:
We are missing the case where parent collection of the dependency is disabled.
We should add a test for that once we have fixed any issues we have there.